File: frontend/lib/models/fuel_record.dart
class FuelRecord {
  final int id;
  final int vehicleId;
  final String? driverId;
  final int? tripId;
  final String fuelType;
  final double quantity;
  final double unitPrice;
  final double totalCost;
  final String currency;
  final int? odometerReading;
  final String? location;
  final double? latitude;
  final double? longitude;
  final String? fuelStationName;
  final String? receiptNumber;
  final DateTime filledAt;
  final DateTime createdAt;

  FuelRecord({
    required this.id,
    required this.vehicleId,
    this.driverId,
    this.tripId,
    required this.fuelType,
    required this.quantity,
    required this.unitPrice,
    required this.totalCost,
    required this.currency,
    this.odometerReading,
    this.location,
    this.latitude,
    this.longitude,
    this.fuelStationName,
    this.receiptNumber,
    required this.filledAt,
    required this.createdAt,
  });

  factory FuelRecord.fromJson(Map<String, dynamic> json) {
    return FuelRecord(
      id: json['id'] as int,
      vehicleId: json['vehicleId'] as int,
      driverId: json['driverId'] as String?,
      tripId: json['tripId'] as int?,
      fuelType: json['fuelType'] as String,
      quantity: (json['quantity'] as num).toDouble(),
      unitPrice: (json['unitPrice'] as num).toDouble(),
      totalCost: (json['totalCost'] as num).toDouble(),
      currency: json['currency'] as String,
      odometerReading: json['odometerReading'] as int?,
      location: json['location'] as String?,
      latitude: json['latitude'] != null 
          ? (json['latitude'] as num).toDouble() 
          : null,
      longitude: json['longitude'] != null 
          ? (json['longitude'] as num).toDouble() 
          : null,
      fuelStationName: json['fuelStationName'] as String?,
      receiptNumber: json['receiptNumber'] as String?,
      filledAt: DateTime.parse(json['filledAt'] as String),
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'vehicleId': vehicleId,
      'driverId': driverId,
      'tripId': tripId,
      'fuelType': fuelType,
      'quantity': quantity,
      'unitPrice': unitPrice,
      'totalCost': totalCost,
      'currency': currency,
      'odometerReading': odometerReading,
      'location': location,
      'latitude': latitude,
      'longitude': longitude,
      'fuelStationName': fuelStationName,
      'receiptNumber': receiptNumber,
      'filledAt': filledAt.toIso8601String(),
      'createdAt': createdAt.toIso8601String(),
    };
  }
}

File: dotnet-backend/projekatFlutter/Models/FuelRecord.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace projekatFlutter.Models
{
    public class FuelRecord:IEntity<int>
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int VehicleId { get; set; }

        [MaxLength(450)]
        public string? DriverId { get; set; }

        public int? TripId { get; set; }

        [Required]
        [MaxLength(30)]
        public string FuelType { get; set; } = string.Empty;

        [Required]
        [Column(TypeName = "decimal(8,2)")]
        public decimal Quantity { get; set; }

        [Required]
        [Column(TypeName = "decimal(8,2)")]
        public decimal UnitPrice { get; set; }

        [Required]
        [Column(TypeName = "decimal(10,2)")]
        public decimal TotalCost { get; set; }

        [MaxLength(3)]
        public string Currency { get; set; } = "BAM";

        public int? OdometerReading { get; set; }

        [MaxLength(255)]
        public string? Location { get; set; }

        [Column(TypeName = "decimal(10,8)")]
        public decimal? Latitude { get; set; }

        [Column(TypeName = "decimal(11,8)")]
        public decimal? Longitude { get; set; }

        [MaxLength(100)]
        public string? FuelStationName { get; set; }

        [MaxLength(50)]
        public string? ReceiptNumber { get; set; }

        [Required]
        public DateTime FilledAt { get; set; } = DateTime.UtcNow;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigation Properties
        [ForeignKey("VehicleId")]
        public Vehicle Vehicle { get; set; } = null!;

        [ForeignKey("DriverId")]
        public User? Driver { get; set; }

        [ForeignKey("TripId")]
        public Trip? Trip { get; set; }
    }
}